sklearn/linear_model/least_angle.py

```diff
@@ -1074,6 +1074,8 @@ def fit(self, X, y):
         """
         self.fit_path = True
         X, y = check_X_y(X, y, y_numeric=True)
+        X = as_float_array(X, copy=self.copy_X)
```

Comment:
Ah, wait, I thought ``y_numeric`` made ``y`` floating, but that is only if it has dtype object. Never mind, sorry for the noise. I didn't have coffee yet.